pkg/v3/types/basetypes.go

```diff
@@ -150,6 +154,36 @@ func (r CheckResult) UniqueID() string {
 	return fmt.Sprintf("%x", resultBytes)
 }
 
+func (r CheckResult) String() string {
+	type CheckResultAlias CheckResult
+	val := &struct {
+		// use all fields from CheckResult
+		CheckResultAlias
+		// override these specific fields
+		UpkeepID    string // base 10 numerical string
+		Trigger     map[string]interface{}
+		PerformData string // 0x-prefixed hex string
+		FastGasWei  string // base 10 numerical string
+		LinkNative  string // base 10 numerical string
+	}{
+		CheckResultAlias: (CheckResultAlias)(r),
+		UpkeepID:         r.UpkeepID.String(),
+		PerformData:      hexutil.Encode(r.PerformData),
+		FastGasWei:       r.FastGasWei.String(),
+		LinkNative:       r.LinkNative.String(),
+	}
```

Comment:
not sure if we should go such a heavy way to create string, we can simply derive this from fields instead of creating a new struct and json marshalling it which is expensive. See uniqueID() function

Reply:
fair, will rework